src/java/org/apache/cassandra/utils/FBUtilities.java

```diff
@@ -1030,6 +1030,22 @@ public static long parseHumanReadableBytes(String value)
         return (long) parseHumanReadable(value, null, "B");
     }
 
+    /**
+     * Parse a double where both a direct value and a percentage are accepted.
+     * For example, for inputs "0.1" and "10%", this function will return 0.1.
+     */
+    public static double parsePercent(String value)
+    {
+        value = value.trim();
+        if (value.endsWith("%"))
```

Comment:
@ethan-brown2022 what if the value ends on multiple "%" like this? "0.1%%%"? Then you will try to parse "0.1%%". It is questionable if we want to do something with it though.

Reply:
If the value ends on multiple "%", it will throw a `NumberFormatException`.  This is the same result as any other string that cannot be parsed as a double, so I think it makes sense for this case as well.

Reply:
ok